### DISCUSSION TITLE
`name` kwarg for routes / path operations incorrectly described as internal only

### DISCUSSION THREAD

**Discussion by silverman63** [fastapi/fastapi#10776] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# example endpoint
@app.get("/{id}", name='details')
def details(request: Request, id: int):
    return { "foo" : "bar" }
```

```html
# example Jinja2 func call
<a href="{{ url_for('details', id=123) }}">Details</a>
```

### Description

Hi all! I noticed in the [docs on Jinja2 templating](hhttps://fastapi.tiangolo.com/advanced/templates/) you mention the ability to use the `url_for` utility function. However, this relies on referencing the `name` kwarg when creating routes. 

When I looked at source code for `@get` decorator, the docstring says: 
```
 Name for this *path operation*. Only used internally.
```

When using Jinja2 url_for, using `name` is a critical component. Could the docs / docstring be updated to reflect this?

### Operating System

macOS

### FastAPI Version

0.104.1

### Pydantic Version

2.5.2

### Python Version

3.11.6

### Additional Context

<img width="748" alt="image" src="https://github.com/tiangolo/fastapi/assets/5976818/449c3ded-e57c-4c33-ada4-e95e225482a1">

**Comment by Kludex** [comment, marked as answer]
PR welcome to include this information. Ping me when ready, please.
